pkg/controller/controller_utils_test.go

```diff
@@ -381,104 +418,161 @@ func TestDeletePodsAllowsMissing(t *testing.T) {
-	pods[8].Status.ContainerStatuses = []v1.ContainerStatus{{RestartCount: 2}, {RestartCount: 1}}
-	pods[8].CreationTimestamp = then
-
-	getOrder := func(pods []*v1.Pod) []string {
-		names := make([]string, len(pods))
-		for i := range pods {
-			names[i] = pods[i].Name
-		}
-		return names
+
+	type podData struct {
+		name              string
+		nodeName          string
+		phase             v1.PodPhase
+		conditions        []v1.PodCondition
+		containerStatuses []v1.ContainerStatus
+		creationTimestamp metav1.Time
 	}
 
-	expected := getOrder(pods)
+	tests := []struct {
+		name          string
+		pods          []podData
+		expectedOrder []string
+	}{
+		{
+			name: "Sorts by active pod",
+			pods: []podData{
+				{name: "unscheduled", nodeName: "", phase: v1.PodPending},
+				{name: "scheduledButPending", nodeName: "bar", phase: v1.PodPending},
+				{name: "unknownPhase", nodeName: "foo", phase: v1.PodUnknown},
+				{name: "runningButNotReady", nodeName: "foo", phase: v1.PodRunning},
+				{name: "runningNoLastTransitionTime", nodeName: "foo", phase: v1.PodRunning,
+					conditions:        []v1.PodCondition{{Type: v1.PodReady, Status: v1.ConditionTrue}},
+					containerStatuses: []v1.ContainerStatus{{RestartCount: 3}, {RestartCount: 0}},
+				},
+				{name: "runningWithLastTransitionTime", nodeName: "foo", phase: v1.PodRunning,
+					conditions:        []v1.PodCondition{{Type: v1.PodReady, Status: v1.ConditionTrue, LastTransitionTime: now}},
+					containerStatuses: []v1.ContainerStatus{{RestartCount: 3}, {RestartCount: 0}},
+				},
+				{name: "runningLongerTime", nodeName: "foo", phase: v1.PodRunning,
+					conditions:        []v1.PodCondition{{Type: v1.PodReady, Status: v1.ConditionTrue, LastTransitionTime: then}},
+					containerStatuses: []v1.ContainerStatus{{RestartCount: 3}, {RestartCount: 0}},
+				},
+				{name: "lowerContainerRestartCount", nodeName: "foo", phase: v1.PodRunning,
+					conditions:        []v1.PodCondition{{Type: v1.PodReady, Status: v1.ConditionTrue, LastTransitionTime: then}},
+					containerStatuses: []v1.ContainerStatus{{RestartCount: 2}, {RestartCount: 1}},
+					creationTimestamp: now,
+				},
+				{name: "oldest", nodeName: "foo", phase: v1.PodRunning,
+					conditions:        []v1.PodCondition{{Type: v1.PodReady, Status: v1.ConditionTrue, LastTransitionTime: then}},
+					containerStatuses: []v1.ContainerStatus{{RestartCount: 2}, {RestartCount: 1}},
+					creationTimestamp: then,
+				},
+			},
+			expectedOrder: []string{
+				"unscheduled",
+				"scheduledButPending",
+				"unknownPhase",
+				"runningButNotReady",
+				"runningNoLastTransitionTime",
+				"runningWithLastTransitionTime",
+				"runningLongerTime",
+				"lowerContainerRestartCount",
+				"oldest",
+			},
+		},
+	}
 
-	for i := 0; i < 20; i++ {
-		idx := rand.Perm(numPods)
-		randomizedPods := make([]*v1.Pod, numPods)
-		for j := 0; j < numPods; j++ {
-			randomizedPods[j] = pods[idx[j]]
+	preparePods := func(podData []podData) []*v1.Pod {
```

Comment:
The benefit of extracting this function, where it is used from only one place is questionable to me. Would it be feasible to have `pods: []*v1.Pod` directly without going via `podData`?

Reply:
I agree with you, there's no need to have an additional function for this. My intention for this first iteration of the refactor was to rewrite the tests to be table-driven while maintaining the coverage.